src/tests/quantuminspire/qiskit/test_qi_result.py

```diff
@@ -95,6 +103,17 @@ def test_more_results(self):
         probabilities = qi_result.get_probabilities()
         self.assertListEqual(probabilities, [{'00': 0.42, '11': 0.58},
                                              {'000': 0.24, '001': 0.25, '010': 0.23, '011': 0.28}])
+
```

Comment:
Typically I would recommend as a best practice to act/assert only once per test method. That way, if the test fails, we know what case specifically is failing. It also allows us to use descriptive names for that particular name to inform the reader what the intended behaviour is that we are trying to assert. Maybe it's a bit big to consider in scope of this PR, since there is already a lot happening, and arguably this is a pre-existing issue, but I would expect a test method to set up a single experiment, and asserting the probabilities once (perhaps using something like [parameterized](https://github.com/wolever/parameterized) to save on boilerplate).

For now, should we at least split out the multiple measurements assertions into a separate test method?

One example of some documentation that recommends this best practice can be found [here](https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#avoid-multiple-acts)

Reply:
In this case it is easy to split it to different tests and name it accordingly.